hudi-common/src/main/java/org/apache/hudi/common/config/DFSPropertiesConfiguration.java

```diff
@@ -117,7 +142,50 @@ public void addProperties(BufferedReader reader) throws IOException {
     }
   }
 
-  public TypedProperties getConfig() {
-    return props;
+  public static TypedProperties getGlobalProps() {
+    final TypedProperties globalProps = new TypedProperties();
+    globalProps.putAll(HUDI_CONF_PROPS);
+    return globalProps;
+  }
+
+  public TypedProperties getProps() {
+    return getProps(false);
+  }
+
+  public TypedProperties getProps(boolean includeHudiConf) {
+    if (includeHudiConf) {
+      TypedProperties mergedProps = new TypedProperties();
+      mergedProps.putAll(HUDI_CONF_PROPS);
+      mergedProps.putAll(props);
+      return mergedProps;
+    } else {
+      return props;
+    }
+  }
+
+  private static Path getDefaultConfPath() {
+    String confDir = System.getenv(CONF_FILE_DIR_ENV_NAME);
+    if (confDir == null) {
+      LOG.warn("Cannot find " + CONF_FILE_DIR_ENV_NAME + ", please set it as the dir of " + DEFAULT_PROPERTIES_FILE);
+      return null;
+    }
+    return new Path("file://" + confDir + File.separator + DEFAULT_PROPERTIES_FILE);
+  }
+
+  private String[] splitProperty(String line) {
+    line = line.replaceAll("\\s+"," ");
+    String delimiter = line.contains("=") ? "=" : " ";
+    int ind = line.indexOf(delimiter);
+    String k = line.substring(0, ind).trim();
+    String v = line.substring(ind + 1).trim();
+    return new String[] {k, v};
+  }
+
+  private boolean isValidLine(String line) {
```

Comment:
add a UT for the regex check?

Reply:
We have some test cases here: https://github.com/apache/hudi/blob/master/hudi-common/src/test/java/org/apache/hudi/common/util/TestDFSPropertiesConfiguration.java#L59-L72. Is it enough or I can add more?